src/battle_script_commands.c

```diff
@@ -16524,6 +16525,22 @@ void BS_TryDefog(void)
     }
 }
 
+void BS_TryUpperHand(void)
+{
+	NATIVE_ARGS(const u8 *failInstr);
+	
+	if (gProtectStructs[gBattlerTarget].obstructed) // Fails if target is obstructed
```

Comment:
Why are you checking this

Reply:
Copied from Sucker Punch's effect; will remove if unnecessary

Reply:
That sounds wrong in both cases